### ISSUE TITLE
Theme broken with Sphinx 8.1.0

### ISSUE THREAD

**Issue by jat255** [jbms/sphinx-immaterial#394]
I updated Sphinx this morning and my docs are no longer building due to a change in the 8.1.0 version of Sphinx released yesterday:

```
Extension error:
Could not import extension sphinx_immaterial.local_mathjax (exception: cannot import name 'MathDomain' from 'sphinx.ext.mathjax' (/lib/python3.12/site-packages/sphinx/ext/mathjax.py))
make: *** [Makefile:20: html] Error 2
```

It looks like the path to `MathDomain` has changed, and the following change to `local_mathjax.py` fixes it:

```diff
$ diff lib/python3.12/site-packages/sphinx_immaterial/local_mathjax.py ~/tmp/local_mathjax.py
7c7,8
< from sphinx.ext.mathjax import MATHJAX_URL, MathDomain
---
> from sphinx.ext.mathjax import MATHJAX_URL
> from sphinx.domains.math import MathDomain
```

The change in Sphinx was in this PR: https://github.com/sphinx-doc/sphinx/pull/12783. I haven't looked into it in detail, but it looks like that `MathDomain` is just being used as a type cast, so maybe there's a better way to solve this. I've pinned Sphinx to 8.0.2 for the time being.

**Comment by 2bndy5** [comment]
See #393

**Comment by jat255** [comment]
Oh, sorry about that! I looked in the issues but didn't look at open PRs, so I missed that it was already being addressed. Thanks for taking care of it!

**Comment by 2bndy5** [comment]
No worries. I recently did the same thing for a different project 🤣 

I'm waiting for the CI to finish publishing v0.12.4 (which includes the fix

**Comment by 2bndy5** [comment]
fix released in [v0.12.4](https://pypi.org/project/sphinx-immaterial/0.12.4/)
